numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -2951,9 +2951,11 @@ _strided_to_strided_multistep_cast(
 
         if (castdata->from.func != NULL) {
             npy_intp out_stride = castdata->from.descriptors[1]->elsize;
+            const char *data[2] = {src, castdata->from_buffer};
```

Comment:
```suggestion
           char *const data[2] = {src, castdata->from_buffer};
```
and below, then I think this should be good go in (i.e. tests pass).